### ISSUE TITLE
Are CGO projects supported?

### ISSUE THREAD

**Issue by peter-mogensen** [rogpeppe/godef#50]
I found out go-mode.el didn't allow me to use godef on exactly those symbols defined in CGO files.
It can be boiled down to that godef will not report this definition:
```go
package test

/*
#include <stdio.h>
*/
import "C"

func Testing() string {
	return "testing"
}
```

However, if the `import "C"` line is included in the comment, the definition is reported correctly.

... using Go 1.7.3
main prg:

```go
package main

import (
	"fmt"
	"test"
)

func main() {
	fmt.Println(test.Testing())
}
```

**Comment by joeblew99** [comment]
@peter-mogensen did you get confirmation on this ?

i have the same issues over here i think:
https://github.com/therecipe/qt/issues/163

**Comment by peter-mogensen** [comment]
No - I didn't  ... however, looking more into this it seems like a very fundamental problem not limited to godef.
However, it's really unfortunate limitation. It would be very nice to be able to navigate to functions defined using CGO.
I seem to recall that the deciding factor is whether there's an "import "C"" statement in the file. So what you could do is to locate all exported API in separate files not using CGO and have them call private functions in files using "import "C""

**Comment by joeblew99** [comment]
thanks for the tip. i wil ask the maintainer over at the QT project.

the others that use the intelli IDEA IDE dont have the issues that VSCode has. they get full intellisense, but they need to devote more RAM to the process for it to work.

**Comment by therecipe** [comment]
There seems to be an open pr that solves this https://github.com/rogpeppe/godef/pull/44

**Comment by rogpeppe** [comment]
This should have been solved by #62.
